opendatahub-io/odh-dashboard
# Default and restrict threshold, add tooltips, default duplicate name and set feature flag

Closes #1383 #1384 

## Description


Do the following things:
1. Set the default value of threshold delta based on the metric type
2. Limit the threshold delta based on the metric type (there will be no error message or helper text, but the creation will be disabled if the user input invalid values)
3.  Add tooltips for all the input fields and descriptions for metrics type dropdown, documentation: https://docs.google.com/document/d/1qq0lobOJQpXD1IowclSWaKhTT_qEPz56IMzvCeOfczY/edit
4. When duplicating the configuration, set the default duplicated name as mentioned in the issue
5. Add `disableBiasMetrics` feature flag in the dashboard config. When the flag is `true`, we do following:
a. Hide the bias tab
b. Disable all the routes related to bias metrics
c. Disabled all the trusty API network calls

## How Has This Been Tested?



To test description 1:
Go to bias metrics configuration modal, change the metric type and see how the threshold delta value changes

To test description 2:
Go to bias metrics configuration modal, try to input some invalid batch size and threshold delta to see if the submit button is disabled

To test description 3:
Go to bias metrics configuration modal, and hover all the tooltips to see if the texts are accurate. Open metrics type dropdown to check the descriptions on the dropdown items

To test description 4:
Duplicate one bias configuration, see if the name field is set as expected

To test description 5:
1. Go to dashboard config CR
2. Set `disableBiasMetrics` to `true`
3. Go to the global model serving page, click on one of the models
4. There should be only the performance tab on the page, no bias tab
5. Change the tab in the URL from `performance` to `bias`, it should navigate you back to `performance`
6. Access the configuration page from the URL, it should navigate you back to `performance`
7. Check the network calls to make sure there is no call for the trusty AI server
8. Repeat every step from 4-7 on the project details page (because they have different URLs), and make sure the results are the same
9. Set Set `disableBiasMetrics` to `false`
10. Wait for 2 mins (or whenever the dashboard re-fetch the dashboard config again)
11. Repeat 3-8, make sure you can use bias metrics as normal

## Test Impact


TBD

## Request review criteria:
